tokio/src/time/driver/sleep.rs

```diff
@@ -170,7 +184,20 @@ impl Sleep {
         let handle = Handle::current();
         let entry = TimerEntry::new(&handle, deadline);
 
-        Sleep { deadline, entry }
+        #[cfg(all(tokio_unstable, feature = "tracing"))]
+        return Sleep {
```

Comment:
Instead of special casing the initialization, you probably could always include a `span` field, but when tracing isn't enabled, it is just a unit struct.